index.js

```diff
@@ -1 +1,151 @@
 // implement your API here
+
+const express = require("express");
+
+// Import in my DB File
+const db = require("./data/db");
+
+const server = express();
+server.use(express.json());
+
+// C U S T O M  M I D D L E W A R E
+function logger(req, res, next) {
+  console.log(
+    `[${new Date().toISOString()}] ${req.method} to ${req.url} ${req.get(
+      "Origin"
+    )}`
+  );
+
+  next();
+}
+
+function atGate(req, res, next) {
+  console.log("At the gate, about to be eaten");
+
+  next();
+}
+
+function auth(req, res, next) {
+  if (req.url === "/mellon") {
+    next();
+  } else {
+    res.send("you shall not pass");
+  }
+}
+
+server.use(logger);
+server.use(atGate);
+
+server.get('/mellon', auth, (req,res) => {
+  console.log('gate opening . . .');
+  console.log('inside and safe');
+  res.send('welcome traveler')
+})
+
+// S E R V E R - R E S P O N S E
+server.get("/", (req, res) => {
+  res.send(`
+  <h1>Hello World (:</h1>
+    <p>Welcome to Web Api Challenege I</p>    
+    `);
+});
+
+// P O S T /api/users
+
+server.post("/api/users", (req, res) => {
+  const dbInfo = req.body;
+  db.insert(dbInfo)
+    .then(user => {
+      res.status(201).json(user);
+    })
+    .catch(err => {
+      res.status(400).json({
+        errorMessage: "Please provide name and bio for the user."
```

Comment:
The catch here will run any time db.insert fails and is not limited to 400 errors. Before db.insert you should have some logic for `if invalid inputs -> res.status(400).json({errormessage...})`